cdbsearch.py

```diff
@@ -73,10 +78,9 @@ def __init__(self, concurrency, evalDecay, cursedWins=False):
 
         # At each level in the tree we need a few threads.
         # Evaluations can happen at any level, so we can saturate the work executor nevertheless
+        # For the root position two threads are probably enough.
         self.executorTree = [
-            concurrent.futures.ThreadPoolExecutor(
-                max_workers=max(2, self.concurrency // 4)
-            )
+            concurrent.futures.ThreadPoolExecutor(max_workers=min(2, self.concurrency))
```

Comment:
min would seem to be wrong  to me, this would limit concurrency. I think the original expression is correct.

Reply:
hm, I am a novice regarding multi-threading. but if the user chooses concurrency 1, should max_workers not be set to 1? and is giving ply==0 so many threads logical? naively I thought it would have less stuff to do, also indicated by your formula self.concurrency // 4

Reply:
the concurrency the user sets only determines the concurrency of the access to cdb. There need to be at least 2 workers in the pool to be able to satisfy that concurrency (at least that would be the case depending on tree shape). With concurrency // 4, I just used some kind of heuristic that seems to allow full concurrency on cdb access, without spawning too many threads on the RPI. 

I'm not sure I understand the ply==0 comment

Reply:
that line also sets the value for `executorTree[0]` right? so when `plu==0`?

Reply:
I understand now the ply==0 comment. It is actually a bit of a bug. I didn't want to have special treatment of ply==0... rather I forgot to have it consistent on line 279... probably one could use just `ThreadPoolExecutor(max_workers=self.concurrency)` as it was not the intention to special case ply 0.

Reply:
in that case let me re-write the patch, could start with empty list unless I am mistaken

Reply:
I have done this now.